tests/ut/ninja.d

```diff
@@ -197,24 +173,16 @@ else
     auto ninja = Ninja(Build(app));
 
     ninja.buildEntries.shouldEqual(
-        [NinjaEntry(buildPath("build gen/protocol.c gen/protocol.h: compiler protocol.proto")),
-         NinjaEntry(buildPath("build bin/protocol.o: gcc gen/protocol.c | gen/protocol.c gen/protocol.h"),
-                    ["before = -o",
-                     "between = -c"]),
-         NinjaEntry(buildPath("build gen/protocol.d: translator gen/protocol.h | gen/protocol.c gen/protocol.h")),
-         NinjaEntry(buildPath("build app: dmd src/main.d bin/protocol.o gen/protocol.d"),
-                    ["before = -of"])
-            ]);
-
-    ninja.ruleEntries.shouldEqual(
-        [NinjaEntry("rule compiler",
-                    ["command = ./compiler $in "]),
-         NinjaEntry("rule gcc",
-                    ["command = gcc $before $out $between $in"]),
-         NinjaEntry("rule translator",
-                    ["command = ./translator $in $out"]),
-         NinjaEntry("rule dmd",
-                    ["command = dmd $before$out $in"])
+        [NinjaEntry(buildPath("build gen/protocol.c gen/protocol.h: _custom_in protocol.proto"),
+                    ["before = ./compiler "]),
+         NinjaEntry(buildPath("build bin/protocol.o: _custom_out  | gen/protocol.c gen/protocol.h"),
+                    ["before = gcc -o ",
+                     "after = $ -c gen/protocol.c"]),
```

Comment:
Previously, the indirect `gen/protocol.c` input seems to have been additionally added as a regular input, and the path replaced as `$in` in the command.